### PR TITLE
Aristo3: CPButton

### PR DESCRIPTION
**Pull request by didierkorthoudt**
Updates to CPButton & NSButton (nib2cib) :
- all styles of buttons are now managed
- buttons can now use material icon images (with automatic color management)
- updated state management (highlight & showState)
- hovered specific image
- Aristo3 CSS theme ready

### CODE REVIEW

**Comment by didierkorthoudt** [comment]
+FIFO

**Comment by didierkorthoudt** [comment]
+FIFO

File: AppKit/CPButton.j [hunk 1]
```diff
@@ -698,87 +688,164 @@ CPButtonImageOffset   = 3.0;
...
+}
+
+- (CPImage)_currentImage
+{
+    var visualState  = [self _imageVisualState],
+        currentImage = [self valueForThemeAttribute:@"image"       inState:visualState],
+        imageColor   = [self valueForThemeAttribute:@"image-color" inState:visualState],
+        buttonIsOn   = ([self state] !== CPOffState);
+
+    if ([currentImage isMaterialIconImage])
```
Comment (mrcarlberg) [inline AppKit/CPButton.j:777]: I can't test this as the method `isMaterialIconImage` is not found

**Comment by mrcarlberg** [comment]
Merged! Thanks!

**Pull request merged**